checkov/terraform/graph_builder/foreach/module_handler.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import annotations
+
+import itertools
+import typing
+from collections import defaultdict
+from copy import deepcopy
+from typing import Any
+
+from checkov.terraform import TFModule
+from checkov.terraform.graph_builder.foreach.abstract_handler import ForeachAbstractHandler
+from checkov.terraform.graph_builder.foreach.consts import FOREACH_STRING, COUNT_STRING
+from checkov.terraform.graph_builder.graph_components.block_types import BlockType
+from checkov.terraform.graph_builder.graph_components.blocks import TerraformBlock
+
+if typing.TYPE_CHECKING:
+    from checkov.terraform.graph_builder.local_graph import TerraformLocalGraph
+
+
+class ForeachModuleHandler(ForeachAbstractHandler):
+    def __init__(self, local_graph: TerraformLocalGraph):
+        super().__init__(local_graph)
+
+    def handle(self, modules_blocks: list[int]) -> None:
+        """
+        modules_blocks (list[int]): list of module blocks indexes in the graph that contains for_each / counts.
+        """
+        if not modules_blocks:
+            return
+        current_level = [None]
+        main_module_modules = deepcopy(self.local_graph.vertices_by_module_dependency.get(None)[BlockType.MODULE])
+        modules_to_render = main_module_modules
+
+        # The for_each rendering will be by modules levels, and not deeper in the graph.
```

Comment:
I think you should extend this documentation a bit, I don't think the word `levels` is understood without the context of a conversation with you

Reply:
perhaps this can even be separated to another function with good documentation